ui/app/src/Utils/FormValidation.elm

```diff
@@ -0,0 +1,62 @@
+module Utils.FormValidation
+    exposing
+        ( ValidationResult(Valid, NotValid)
+        , ValidatedString
+        , validateDate
+        , areValidResults
+            , validatedMatcherToMatcher
+        , ValidatedMatcher
+        )
+
+import Utils.Date exposing (timeFromString)
+import Silences.Types exposing (Silence)
+import Utils.Types exposing (Matcher)
+
+
+type ValidationResult
+    = Valid
+    | NotValid String
+
+
+type alias ValidatedString =
+    { value : String
+    , result : ValidationResult
+    }
```

Comment:
I would need to play with this / ask for the wisdom of the @w0rm, but potentially it could also be

```elm
type ValidationResult a
    = Valid a
    | NotValid String
```
I think then it can capture different types, like when using a `Maybe`. So you would have a function signature like:

```elm
isMatcherValid : ValidationResult Matcher -> Bool
```